theme/views.py

```diff
@@ -30,11 +31,23 @@ def get_context_data(self, **kwargs):
             except:
                 u = User.objects.get(username=self.request.GET['user'])
 
-        res = BaseResource.objects.filter(user=u)
+        # get all resources the profile user owns
+        resources = u.uaccess.get_owned_resources()
+
+        # if requesting user is not the profile user, then show only resources that the requesting user has access
+        if self.request.user != u:
+            if self.request.user.is_authenticated():
+                # filter out any resources the requesting user doesn't have access
+                resources = resources.filter(Q(pk__in=self.request.user.uaccess.get_held_resources()) |
```

Comment:
It should not do two queries. You can filter based on join of another QuerySet and until you actually count or enumerate over that queryset it will not be executed, so only the final combined queries will be run as long as you are careful.